<launch>

  <arg name="exclude_sim" default="false" />
  <arg name="exclude_nav" default="false" />
  <arg name="exclude_rviz" default="false" />
  <arg name="exclude_goals" default="false" />

  <arg name="output_files_dirpath" default="$(env HOME)/images/"/>


  <!-- Simulation -->
  <include file="$(find infobot_uw_morse)/launch/simple.launch" unless="$(arg exclude_sim)">
    <arg name="use_joystick" value="false" />
  </include>

  <!-- Navigation -->
  <include file="$(find infobot_uw_systems)/launch/navigation.launch" unless="$(arg exclude_nav)">
    <arg name="map_name" value="sim-simple" />
  </include>

  <!-- Rviz -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find infobot_find_viewpoint)/test/find_viewpoint.rviz" unless="$(arg exclude_rviz)"/>

  <!-- FindViewpoint -->
  <include file="$(find infobot_find_viewpoint)/launch/find_viewpoint.launch">
    <arg name="pmap_files_dirpath" value="$(find infobot_uw_world)/pmaps" />
    <arg name="octomap_files_dirpath" value="$(find infobot_uw_world)/octomaps" />
    <arg name="cloud_in" value="/xtion/depth_registered/points" />
    <arg name="topomap_files_dirpath" value="$(find infobot_uw_world)/topomaps" />

    <arg name="image_topic" value="/prosilica_vga/image_rect_color"/>
    <arg name="output_files_dirpath" value="$(arg output_files_dirpath)"/>

    <arg name="fov_grid_res" value="50"/>
    <arg name="dist_factor" value="1"/>

    <arg name="rect_filter_margin" default="1.5"/>
    <arg name="camera_trans_z" default="1.314"/>
    <arg name="max_iter" default="5"/>
    <arg name="max_move_base_iter" default="5"/>
    <arg name="move_base_filter_margin" default="2.0"/>
  </include>

  <!-- FindViewpoint Goals -->
  <node pkg="infobot_find_viewpoint" type="send_multiple_goals" name="send_multiple_goals" unless="$(arg exclude_goals)" output="screen">
    <param name="mode" value="0"/>
    <param name="octomap_filename" value="sim-simple-model.bt"/>
    <param name="topomap_filename" value="sim-simple_corridor-topo.yaml"/>
    <rosparam param="pmap_filenames">[
    "sim-simple-room1.yaml",
    "sim-simple-room2.yaml",
    "sim-simple-room3.yaml"
    ]</rosparam>
  </node>

</launch>
